h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ParquetParser.java

```diff
@@ -73,15 +73,25 @@ public static ParseSetup guessSetup(ByteVec vec, byte[] bits) {
     }
     // seems like we have a Parquet file
     byte[] metadataBytes = VecParquetReader.readFooterAsBytes(vec);
```

Comment:
Ins't it better to put all these into the second method (guessDataSetup)? Since you want to separate check for parquet file from processing the setup further, wouldn't it be better to only check the magic bits here and do all the rest in the second method with unchecked exception?

Reply:
That would be better but the file can still be a CSV file if it starts with the magic bytes ("PAR1"). This is why I try to parse the whole metadata to make sure.